### PR TITLE
remove minor versions for tomcat

### PR DESCRIPTION
**Pull request by dirkmueller**
there should be "branch" (which is major version) and fullversion-flavor.

### CODE REVIEW

File: src/bci_build/package/apache_tomcat.py [hunk 1]
```diff
@@ -91,19 +90,14 @@ def _get_sac_supported_until(
...
-            ),
         ],
         cmd=[
             f"/usr/{'libexec' if os_version in (OsVersion.TUMBLEWEED, OsVersion.SLE16_0) else 'lib'}/tomcat/server",
             "start",
         ],
         exposes_tcp=[8080],
         env={
-            "TOMCAT_MAJOR": tomcat_major,
+            "TOMCAT_MAJOR": math.floor(tomcat_ver),
```
Comment (alexandrevicenzi) [inline src/bci_build/package/apache_tomcat.py:100]: nit: `int(tomcat_ver)` achieves the same thing and does not require an extra library import.

File: src/bci_build/package/apache_tomcat.py [hunk 2]
```diff
@@ -121,12 +115,12 @@ def _get_sac_supported_until(
         entrypoint_user="tomcat",
         logo_url="https://tomcat.apache.org/res/images/tomcat.png",
     )
-    for tomcat_major, os_version, jre_version in (
-        (10, OsVersion.TUMBLEWEED, 22),
-        (10, OsVersion.TUMBLEWEED, 21),
-        (10, OsVersion.TUMBLEWEED, 17),
+    for tomcat_ver, os_version, jre_version in (
```
Comment (alexandrevicenzi) [inline src/bci_build/package/apache_tomcat.py:118]: if you are hardcoding versions already, why not include `tomcat_ver_major` and avoid any math conversions?

**Review by dcermak — COMMENTED** [review]
Floats are imho not the right way to handle version numbers due to potential representation & rounding issues. I'd suggest to use either strings or instances of [`packaging.Version`](https://packaging.pypa.io/en/stable/version.html)

File: src/bci_build/package/apache_tomcat.py (hunk 2 above)
Comment (dirkmueller) [reply src/bci_build/package/apache_tomcat.py:118]: deriving it from one value is more DRY than duplicating it in the tuple, imho.

File: src/bci_build/package/apache_tomcat.py [hunk 3]
```diff
@@ -6,18 +6,17 @@
...
 
-_TOMCAT_VERSIONS: list[int] = [9, 10]
-assert _TOMCAT_VERSIONS == sorted(_TOMCAT_VERSIONS)
+# last version needs to be the newest
+_TOMCAT_VERSIONS: list[str] = ["9", "10.1"]
 
 
 def _get_sac_supported_until(
-    os_version: OsVersion, tomcat_major: int, jre_major: int
+    os_version: OsVersion, tomcat_major: str, jre_major: int
```
Comment (dcermak) [inline src/bci_build/package/apache_tomcat.py:19]: nit: this is no longer only the major version (needs more adjustment down the line too)
```suggestion
    os_version: OsVersion, tomcat_ver: str, jre_major: int
```

**Review by dcermak — COMMENTED** [review]
LGTM, only have one nitpick

File: src/bci_build/package/apache_tomcat.py (hunk 3 above)
Comment (dirkmueller) [reply src/bci_build/package/apache_tomcat.py]: all references renamed.

**Pull request merged**